internal/common/model.go

```diff
@@ -1,17 +1,43 @@
 package common
 
-import "github.com/razorpay/metro/internal/app"
+import (
+	"fmt"
+
+	"github.com/razorpay/metro/internal/app"
+)
 
 // IModel interface which all models should implement
 type IModel interface {
 	// Key returns the key to store the model against
 	Key() string
 	// Prefix returns the key prefix used by the module
 	Prefix() string
+	// Returns the version of the saved model in the registry. Version changes for every update.
+	GetVersionID() (string, error)
+	//Sets the version of the saved model from the registry. Never to be used by business logic.
+	SetVersionID(vid string)
```

Comment:
This would mean that setting the version id should happen in the same package where `BaseModel` is defined? But version ID has to be set in the `repo` of each domain. Am I missing something?

Reply:
This will happen inside the `Get()` of `common/repo.go`

Reply:
Created Issue #210 to handle this